src/vaadin-text-field-mixin.html

```diff
@@ -523,6 +555,12 @@
       }
     }
 
+    /**
+     * @param {boolean?} required
+     * @param {number?} minlength
+     * @param {number?} maxlength
+     * @param {string?} maxlength
+     */
     _constraintsChanged(required, minlength, maxlength, pattern) {
```

Comment:
Looks like this is missing `@protected`? Also `?` suffix doesn't mean anything in JSDoc. `?` as a prefix would be nullable type and `=` as a suffix would be optional parameter.
```suggestion
    /**
     * @param {boolean | undefined} required
     * @param {number | undefined} minlength
     * @param {number | undefined} maxlength
     * @param {string | undefined} maxlength
     * @protected
     */
    _constraintsChanged(required, minlength, maxlength, pattern) {
```

Reply:
`=` suffix could also work here, but we'd need to also annotate the private override of `_constraintsChanged` in number-field with optional params since it adds a new param at the end and otherwise the generator complains that you can't have a non-optional parameter after optional params.

Reply:
Thanks, changed.